stan/math/prim/fun/gaus_interp.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef STAN_MATH_PRIM_FUN_GAUS_INTERP
+#define STAN_MATH_PRIM_FUN_GAUS_INTERP
+
+#include <stan/math/prim/fun/conv_gaus_line.hpp>
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+namespace stan {
+namespace math {
+
+// set tolerance for how close interpolated values need to be to
+// the specified function values
+const double INTERP_TOL = 1e-8;
+const double SIG2_SCALE = 0.1;
+
+/*
+given a set of points (x_i, y_i) with x_i's in increasing order, find
+a_i, b_i such that the line between (x_i, y_i) and (x_{i+1}, y_{i+1}) is
+f(t) = a_i*t + b_i, store the a_i's and b_i's in as and bs.
+*/
+void lin_interp_coefs(int n, std::vector<double> xs, std::vector<double> ys,
+                      std::vector<double>& as, std::vector<double>& bs) {
```

Comment:
I'm generally against in/out parameters and I think this function's definition could just be inlined where it's used

Reply:
It has been inlined